subworkflows/local/long_reads_qc.nf

```diff
@@ -0,0 +1,95 @@
+include { FASTP as FASTP_LR                       } from '../../modules/nf-core/fastp/main'
+include { RAW_READ_QUALITY_CHECK                  } from '../../modules/local/raw_read_quality_check/'
+include { MINIMAP2_ALIGN as HUMAN_DECONTAMINATION } from '../../modules/nf-core/minimap2/align/main'
+include { MINIMAP2_ALIGN as HOST_DECONTAMINATION  } from '../../modules/nf-core/minimap2/align/main'
+
+workflow LONG_READS_QC {
+
+    take:
+    reads                   // [ val(meta), path(reads) ]
+    reference_genome   // [ val(meta2), path(reference_genome) ]
+
+    main:
+    ch_versions = Channel.empty()
+
+    FASTP_LR(
+        reads,
+        [],
+        false,
+        false,
+        false,
+        false
+    )
+
+    ch_versions = ch_versions.mix(FASTP_LR.out.versions)
+
+    RAW_READ_QUALITY_CHECK(
```

Comment:
I would remove this one with in favour of json splirping and the branch operator.

```groovy
quality_levels_ch = FASTP_LR.out.json.map { meta, json -> {
        json_txt = new JsonSlurper().parseText(json.text)
        q20bases = json_txt?.summary?.before_filtering?.q20_bases ?: 0;
        total_bases = json_txt?.summary?.before_filtering?.total_bases ?: 0;

        q20_percentage = q20_bases / total_bases * 100

        categorization = [
            "high_quality": q20_percentage >= 80, 
            "low_quality": q20_percentage < 80,
        ]
        return [meta, categorization]
    } 
}
```
It's also worth discarding data that was heavily filtered, like we do for short reads.